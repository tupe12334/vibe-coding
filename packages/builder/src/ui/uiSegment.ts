import type { ListItem, RootContent } from "mdast";
import { uiRules } from "./rules";

export const uiSegment = async (): Promise<RootContent[]> => {
  const segment: RootContent[] = [
    {
      type: "heading",
      depth: 2,
      children: [{ type: "text", value: "General UI Guidelines" }],
    },
    ...uiRules.map(
      (rule): ListItem => ({
        type: "listItem",
        children: [
          { type: "paragraph", children: [{ type: "text", value: rule }] },
        ],
      })
    ),
  ];
  return segment;
};
