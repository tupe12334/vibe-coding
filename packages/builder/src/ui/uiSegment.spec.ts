import { describe, expect, test } from "vitest";
import { uiSegment } from "./uiSegment";

describe("uiSegment", () => {
  test("returns general ui guidelines", async () => {
    const segment = await uiSegment();
    expect(segment).toMatchSnapshot();
  });
});
