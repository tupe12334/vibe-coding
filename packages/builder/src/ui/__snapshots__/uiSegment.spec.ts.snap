// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`uiSegment > returns general ui guidelines 1`] = `
[
  {
    "children": [
      {
        "type": "text",
        "value": "General UI Guidelines",
      },
    ],
    "depth": 2,
    "type": "heading",
  },
  {
    "children": [
      {
        "children": [
          {
            "type": "text",
            "value": "Write styling in a direction agnostic way: i.e. use start and end instead of left and right",
          },
        ],
        "type": "paragraph",
      },
    ],
    "type": "listItem",
  },
]
`;
